binder/src/main/java/io/grpc/binder/internal/ActiveTransportTracker.java

```diff
@@ -0,0 +1,114 @@
+package io.grpc.binder.internal;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.HashSet;
+import javax.annotation.concurrent.GuardedBy;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.internal.ServerStream;
+import io.grpc.internal.ServerTransportListener;
+
+/**
+ * Tracks which {@link BinderTransport.BinderServerTransport} are currently active and allows
+ * invoking a {@link BinderTransportSecurity.TerminationListener} only once all transports are
+ * terminated.
+ */
+final class ActiveTransportTracker {
+  @GuardedBy("this")
+  private final HashSet<BinderTransport.BinderServerTransport> activeTransports = new HashSet<>();
```

Comment:
What are the pros/cons of tracking this as a set vs just counting the number of transports that have been created but not yet terminated?

Reply:
Using a set would guarantee that whatever transport is terminating is indeed getting tracked. That would protect this class against mishandling:

- the termination of a transport that was never tracked to begin with
- duplicate termination notifications for the same transport

It might be just my paranoia though. Changed to tracking just the count of active transports for now.